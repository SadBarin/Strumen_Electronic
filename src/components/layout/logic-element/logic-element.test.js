import LogicElement from "./index";

const elementProp = {id: 1, logic: "AND"};

describe("Transfer props check for LogicElement", () => {
  const element = (props) => mount(<LogicElement {...props} />);
  let logicElement;

  beforeEach(() => {
    logicElement = element(elementProp);
  });

  it("Transfer props id check for LogicElement", () => {
    expect(logicElement.props().id).toEqual(elementProp.id);
  });

  it("Transfer props logic check for LogicElement", () => {
    expect(logicElement.props().logic).toEqual(elementProp.logic);
  });
})

it("LogicElement snapshot check", () => {
  const logicElementTree =
    render(<LogicElement id={elementProp.id} logic={elementProp.logic} />);

  expect(toJson(logicElementTree)).toMatchSnapshot();
});
