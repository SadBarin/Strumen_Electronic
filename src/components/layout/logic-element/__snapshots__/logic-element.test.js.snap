// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogicElement snapshot check 1`] = `
<div
  class="position-absolute logic-element-container"
>
  <div
    class="logic-element react-draggable"
    style="transform:translate(undefinedpx,undefinedpx)"
  >
    <div
      class="circle circle-left"
    />
    <div
      class="handle"
    />
    <select
      class="form-select logic-element-select"
    >
      <option
        value="OR"
      >
        OR
      </option>
      <option
        selected=""
        value="AND"
      >
        AND
      </option>
      <option
        selected=""
        value="AND"
      >
        NOT
      </option>
      <option
        selected=""
        value="AND"
      >
        NOR
      </option>
      <option
        value="NAND"
      >
        NAND
      </option>
      <option
        value="XOR"
      >
        XOR
      </option>
      <option
        value="XNOR"
      >
        XNOR
      </option>
    </select>
    <div
      class="circle circle-right"
    />
  </div>
</div>
`;
